src/main/java/build/buildfarm/instance/shard/DistributedState.java

```diff
@@ -28,28 +28,20 @@
 public class DistributedState {
   /**
    * @field executeWorkers
-   * @brief All of the execute-only workers register themselves to the cluster.
+   * @brief All of the execute-capable workers register themselves to the cluster.
    * @details This is done to keep track of which machines are online and known by the rest of the
    *     cluster.
    */
   public RedisHashMap executeWorkers;
 
   /**
    * @field storageWorkers
-   * @brief All of the storage-only workers register themselves to the cluster.
+   * @brief All of the storage-capable workers register themselves to the cluster.
    * @details This is done to keep track of which machines are online and known by the rest of the
    *     cluster.
    */
   public RedisHashMap storageWorkers;
```

Comment:
To echo a previous comment from https://github.com/bazelbuild/bazel-buildfarm/pull/1152,
We may want to store all the workers together and filter as needed based on capabilities.  i.e. reduce this down even further to a single HashMap.  Not necessary to change now, but throwing it out as a suggestion.